src/overlays/ovl_i2/fox_me.c

```diff
@@ -1,97 +1,2527 @@
+
+                    boss->swork[0] = 0;
+
+                    Matrix_RotateZ(gCalcMatrix, (boss->obj.rot.z - 45.0f) * M_DTOR, 0);
+
+                    src.x = 0.0f;
+                    src.y = 156.0f;
+                    src.z = 930.0f;
+
+                    Matrix_MultVec3fNoTranslate(gCalcMatrix, &src, &dest);
+
+                    func_i2_80188298(boss->obj.pos.x + dest.x, boss->obj.pos.y + dest.y, boss->obj.pos.z + dest.z,
+                                     boss->obj.rot.z - 45.0f);
+                    func_i2_80188228(boss->obj.pos.x - dest.x, boss->obj.pos.y - dest.y, boss->obj.pos.z + dest.z,
+                                     boss->obj.rot.z - 45.0f);
+                    func_i2_80188228(boss->obj.pos.x + dest.y, boss->obj.pos.y - dest.x, boss->obj.pos.z + dest.z,
+                                     (boss->obj.rot.z - 45.0f) + 90.0f);
+                }
+            }
+            break;
+
+        case 3:
+            boss->timer_050 = 30;
+            boss->actionState = 4;
+            boss->timer_05C = 15;
+            boss->swork[0] = 0;
+            break;
+
+        case 4:
+            if (boss->timer_050 == 0) {
+                func_8007D2C8(boss->obj.pos.x, boss->obj.pos.y, boss->obj.pos.z + 500.0f, 30.0f);
+                func_8007BFFC(boss->obj.pos.x, boss->obj.pos.y, boss->obj.pos.z + 500.0f, 0.0f, 0.0f, 0.0f, 20.0f, 30);
+                boss->actionState = 5;
+
+                boss->info.unk_10 = 1000.0f;
+
+                func_8001A55C(&boss->sfxPos, 0x3103605B);
+                Audio_PlaySfx(0x3102405DU, &boss->sfxPos, 4U, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+            }
+            break;
+
+        case 5:
+            boss->obj.rot.z -= 3.0f;
+            if (boss->vel.z < 5.0f) {
+                boss->vel.z = boss->vel.z + 1.0f;
+            }
+            break;
+    }
+
+    if (boss->unk_078.z >= 360.0f) {
+        boss->unk_078.z = boss->unk_078.z - 360.0f;
+    }
+    if (boss->unk_078.z < 0.0f) {
+        boss->unk_078.z += 360.0f;
+    }
+}
+
+void func_i2_801887D0(Boss* boss) {
+    s32 i;
+
+    if (boss->timer_054 & 1) {
+        RCP_SetupDL_27();
+        gDPSetPrimColor(gMasterDisp++, 0, 0, 0, 255, 0, 255);
+    }
+
+    Matrix_RotateZ(gGfxMatrix, (M_PI / 4), 1);
+    Matrix_SetGfxMtx(&gMasterDisp);
+    gSPDisplayList(gMasterDisp++, D_60240B0);
+    RCP_SetupDL_64();
+
+    for (i = 1; i < 4; i++) {
+        if (boss->swork[0] != 0) {
+            gDPSetPrimColor(gMasterDisp++, 0, 0, 255, 255, 255, boss->swork[0]);
+            Matrix_Push(&gGfxMatrix);
+            Matrix_RotateZ(gGfxMatrix, i * (M_PI / 2), 1);
+            Matrix_Translate(gGfxMatrix, 0.0f, 156.0f, 930.0f, 1);
+
+            if (gGameFrameCount & 1) {
+                Matrix_Scale(gGfxMatrix, 6.0f, 9.0f, 5.0f, 1);
+            } else {
+                Matrix_Scale(gGfxMatrix, 4.0f, 7.0f, 5.0f, 1);
+            }
+
+            Matrix_RotateZ(gGfxMatrix, 1.5707964f, 1); // (M_PI / 2) does not match
```

Comment:
Are you sure? VSCode says they are equal. Maybe 2 vs 2.0f makes a difference?

Reply:
Nope, it doesn't match with either, it's very weird